app-modules/case-management/src/Filament/Resources/CaseResource/Pages/Concerns/HasCaseHeader.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\CaseManagement\Filament\Resources\CaseResource\Pages\Concerns;
+
+use AdvisingApp\Prospect\Filament\Resources\ProspectResource;
+use AdvisingApp\Prospect\Models\Prospect;
+use AdvisingApp\StudentDataModel\Filament\Resources\StudentResource;
+use AdvisingApp\StudentDataModel\Models\Student;
+use Illuminate\Contracts\View\View;
+
+trait HasCaseHeader
+{
+    public function getHeader(): ?View
+    {
+        $from = app('router')->getRoutes()->match(app('request')->create(url()->previous()))->getName();
+
+        if ($from === 'filament.admin.resources.students.view' && $this->record->respondent instanceof Student) {
+            $backButtonLabel = 'Back to student';
+            $backButtonUrl = StudentResource::getUrl('view', ['record' => $this->record->respondent->getKey()]);
+        } elseif ($from === 'filament.admin.resources.prospects.view' && $this->record->respondent instanceof Prospect) {
+            $backButtonLabel = 'Back to prospect';
+            $backButtonUrl = ProspectResource::getUrl('view', ['record' => $this->record->respondent->getKey()]);
+        }
```

Comment:
@danharrin what are your thoughts on this? Rather than looking at what the previous URL is I would probably recommend adding the ability to read from a URL param what to redirect back to and the label. So it can be controlled at the link level and could be re-used between Case pages.